### PR TITLE
GH4255: Introduce Verify for CodeGen Expectations

### LINKED ISSUE #4255: Introduce Verify for CodeGen Expectations / Asserts

### PR DESCRIPTION
**Pull request by devlead**
* fixes #4255

### CODE REVIEW

**Comment by SimonCropp** [comment]
u should tweak encoding and line endings in gitattributes and editor config

**Comment by SimonCropp** [comment]
if builds fail on CI it is nice to expose the recieved files for download https://github.com/VerifyTests/Verify/blob/main/docs/build-server.md

File: src/Cake.Core.Tests/VerifyConfig.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+using VerifyXunit;
+
+namespace Cake.Core.Tests
+{
+    public static class VerifyConfig
+    {
+        [ModuleInitializer]
+        public static void Init()
+        {
+            Verifier.DerivePathInfo(Expectations.Initialize);
```
Comment (SimonCropp) [inline src/Cake.Core.Tests/VerifyConfig.cs:12]: if u want to have `.verified.cake` as snapshot files then `EmptyFiles.FileExtensions.AddTextExtension("cake");` will make it easier

**Comment by devlead** [comment]
> if builds fail on CI it is nice to expose the recieved files for download https://github.com/VerifyTests/Verify/blob/main/docs/build-server.md

Sensible addition can add that in dotnet test Task OnError
https://cakebuild.net/docs/writing-builds/tasks/error-handling

I'll add an issue to track that post 4.0 release.

**Comment by devlead** [comment]
> u should tweak encoding and line endings in gitattributes and editor config

added.

**Comment by devlead** [comment]
> Verify.DiffPlex help with the signal to noise. especially in CI https://github.com/VerifyTests/Verify.DiffPlex

I'll add an issue to track that post 4.0 release.

**Review by patriksvensson — APPROVED** [review]
LGTM! 👍

**Pull request merged**